src/kernel_abi.h

```diff
@@ -2103,6 +2103,9 @@ struct ARM64Arch : public GenericArch<SupportedArch::aarch64, WordSize64Defs> {
   };
   typedef struct user_pt_regs user_regs_struct;
 
+#if defined (__i386__)
+  typedef long double __uint128_t;
+#endif
```

Comment:
This case is a bit more painful.

`long double` isn't right. I think instead something like
```
typedef struct __uint128_t {
  uint64_t parts[2];
};
```